Update python Assignment

hareem211 opened the pull request.

hareem211 committed:
Update python Assignment

ifzal commented:
### Formatting guidelines
#### 1. Purpose of this PR
#### 2. What is the problem statement/question being addressed?
#### 3. How and what changes are made to fix/solve the problem?
#### 4. Targeted environment setup for running/testing this solution?
#### 5. How to run this solution and what should be the outcome?

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+import csv
```
Why this is not a `.py` file? 
Titlecase shouldn't be used for file naming conventions.

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+import csv
 
+def read_weather_data(weather_data):
```
```suggestion
def read_weather_data(weather_data_file_path):
```
or 
```
def read_weather_data(input_file_path):
```

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+            min_pressure_9am = pressure_9am
+
+    return min_pressure_9am
+file_path = r"C:\Users\ue\Desktop\weatherAUS\weatherAUS.csv"
```
@hareem211 Don't hardcode this path variable contents, instead user `relative path` and path should be provided by user as a parameter to run this code.

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
 
+def read_weather_data(weather_data):
+    weather_data = []
+    with open( r"C:\Users\ue\Desktop\weatherAUS\weatherAUS.csv") as csvfile:
```
@hareem211 Why you have used with operator?

hareem211 replied:
with operator is used when dealing with file...I've used with operator because it completely closed the file after its suite finishes

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+        for row in reader:
+            weather_data.append(row)
+        return weather_data
+def calculate_avg_temperatures(weather_data):
```
@hareem211 Study what is main function in python code files and what's purpose of that function?

hareem211 replied:
In Python, the role of main function is to act as the starting point of execution for any software program. The execution of the program starts only when the main function is defined in python because the program executes only when it runs directly.

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+
+avg_min_temp, avg_max_temp = calculate_avg_temperatures(weather_data)
+highest_rainfall_each_year = find_highest_rainfall_each_year(weather_data)
+min_pressure_9am = find_min_pressure_at_9am(weather_data)
```
@hareem211 What if I need values for other hours too? would you write more functions to do that for you?

hareem211 replied:
no...rather I'll change the hour time with the hour I needed

ifzal commented on Assignment:
```diff
@@ -1 +1,83 @@
+file_path = r"C:\Users\ue\Desktop\weatherAUS\weatherAUS.csv"
+weather_data = read_weather_data(file_path)
+
+avg_min_temp, avg_max_temp = calculate_avg_temperatures(weather_data)
```
@hareem211 This should require user to input the month for which calculations are rquired.

hareem211 committed:
WeatherAUS

1. Purpose of this PR
          The purpose of this PR is to calculate the average minimum and maximum temperatures by month, the highest rainfall of each year, and the minimum pressure at 9 AM from the weatherAUS.csv file.
2. What is the problem statement/question being addressed?
           The problem statement is to find the average minimum and maximum temperatures by month, the highest rainfall of each year, and the minimum pressure at 9 AM from the weatherAUS.csv file.
3. How and what changes are made to fix/solve the problem?
           The changes made to fix/solve the problem are as follows:
    * A new function, calculate_avg_temperatures, is created to calculate the average minimum and maximum temperatures by month.
    * A new function, find_highest_rainfall_each_year, is created to find the highest rainfall of each year.
   * A new function, find_min_pressure_at_9am, is created to find the minimum pressure at 9 AM.
4. Targeted environment setup for running/testing this solution?
            The targeted environment setup for running/testing this solution is windows , a Linux or macOS system with Python 3.6 or higher installed.
5. How to run this solution and what should be the outcome?
            To run this solution, follow these steps:
1. Clone the repository to your local machine.
2. Open a terminal window and navigate to the directory where you cloned the repository.
3. Run the following command:
         python main.py
​This will print the average minimum and maximum temperatures by month, the highest rainfall of each year, and the minimum pressure at 9 AM from the weatherAUS.csv file.
